data-prepper-core/src/main/java/com/amazon/dataprepper/peerforwarder/PeerForwarderConfiguration.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.peerforwarder;
+
+import com.amazon.dataprepper.peerforwarder.discovery.DiscoveryMode;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.AssertTrue;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+/**
+ * Class to hold configuration for Core Peer Forwarder in {@link com.amazon.dataprepper.parser.model.DataPrepperConfiguration},
+ * including port, ssl, buffer, peer forwarder client and server configuration.
+ * @since 2.0
+ */
+
+public class PeerForwarderConfiguration {
+    private Integer serverPort = 21890;
+    private Integer requestTimeout = 10_000;
```

Comment:
We should move to using `Duration`. Data Prepper plugins have a built-in configuration for `Duration` values that supports ISO-8601 durations and values like `1s` or `10s`. If this were a plugin, it would work automatically.

We may want to find a way to support the `PluginDurationDeserializer` class here as well. It is probably beyond of the scope of this PR, but we should track this change.